### DISCUSSION TITLE
Phonon dispersion in a single-layer material

### DISCUSSION THREAD

**Discussion by teguh2605** [ttadano/alamode#237] [Q&A]
Dear Tadano,

I'm trying to reproduce the phonon dispersion of a single-layer 1T-FeCl2. However, I always obtained negative energy in phonon dispersion, which is incorrect. Before that, I always received a fitting error of force constant above 15%. My question is, does the high error influence the negative energy? if not, could you give a tip on how to remove the negative energy?

Regards,

Teguh

**Comment by ttadano** [comment]
Hi,

Can you tell us more specifically where in the Brillouin zone the negative mode appears?

**Reply by teguh2605** [reply]
![Figure_1](https://github.com/user-attachments/assets/c9f573e7-b723-44ec-af45-b59902fa8067)
Dear Tadano,

I attach the figure and here is my input file 

&general
 PREFIX = FeCl2
 MODE = suggest
 NAT = 12; NKD = 2
 KD = Fe Cl
TOLERANCE = 1.0e-4
NONCOLLINEAR = 1
MAGMOM = 0 0 1 0 0 1 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
/

&interaction
 NORDER = 1
/

&optimize
 &DFSET = DFSET_harmonic
&/

&cutoff
 *-* None 
/

&cell
1.8897259886
6.9600000381         0.0000000000         0.0000000000
   -3.4800000191         6.0275368434         0.0000000000
   0 0 17.260000000000
/

&position
1 0.000000000         0.000000000         0.000000000
1 0.000000000         0.500000000         0.000000000
1 0.500000000         0.000000000         0.000000000
1 0.500000000         0.500000000         0.000000000
2 0.166665003         0.333335012         0.079930000
2 0.166665003         0.833335042         0.079930000
2 0.666665018         0.333335012         0.079930000
2 0.666665018         0.833335042         0.079930000
2 0.333335012         0.166665003         0.920069993
2 0.333335012         0.666665018         0.920069993
2 0.833335042         0.166665003         0.920069993
2 0.833335042         0.666665018         0.920069993
/

I set the magnitude of displacement of 10^-5 with the supercell of 2x2, Also, I got the fitting error of 99% even setting --offset.
I'm very thankful if you give suggestions to solve the problem.

Regards,


Teguh

**Reply by ttadano** [reply]
Have you optimized the atomic positions using the supercell? If not, doing so may resolve the issue.

**Reply by teguh2605** [reply]
Did you mean the original supercell or after displacement?

**Reply by ttadano** [reply]
The original supercell before displacements.

**Reply by teguh2605** [reply]
Dear Tadano,

I optimized the original supercell as suggested, but the negative energy still exists (please find the attached file). Is there any suggestion? 

Regards,


Teguh

![Figure_1](https://github.com/user-attachments/assets/df19b45f-c5cd-49f5-b236-3cac48d5c01c)

**Reply by teguh2605** [reply]
Dear Tadano,

I found a confusing case. Previously, the displce.py suggests 6 displacement, as I found in the pattern_HARMONIC file. However, as 6 displacement outputs (disp1.out, ..., disp6.out) were finished, the extract.py always exclude disp1.out as written in the DFSET_harmonic file. even I remove one of the remaining outputs. does the program choose the appropriate ones? because my early calculations include all the outputs.

Regards,


Teguh
